django/db/models/base.py

```diff
@@ -1063,6 +1063,8 @@ def _perform_unique_checks(self, unique_checks):
                 continue
 
             qs = model_class._default_manager.filter(**lookup_kwargs)
+            if condition:
+                qs = qs.filter(condition)
```

Comment:
That's too naive. It has to consider whether or not `self` matches the condition in the first place.

Reply:
removed this check at all, left only db constraint check